shm/player.cpp

```diff
@@ -0,0 +1,29 @@
+#include "player.hpp"
+
+Player::Player(std::shared_ptr<Ship> ship, size_t money, size_t availableSpace)
+    : ship_(ship)
+    , money_(money)
+    , availableSpace_(availableSpace)
+{}
+
+std::shared_ptr<Ship> Player::getShip() const {
+    return ship_;
+}
+
+size_t Player::getMoney() const {
+    return money_;
+}
+
+size_t Player::getAvailableSpace() const {
+    return availableSpace_;
+}
+
+size_t Player::getSpeed() const {
+    return ship_->getSpeed();
+}
+
+std::shared_ptr<Cargo> Player::getCargo(size_t index) const {
+    std::shared_ptr<Cargo> cargoPtr;
+    cargoPtr = ship_->getCargo().at(index);
```

Comment:
What if ptr is nullptr? This function is not safe! You need to validate if you can access some cargo. Try `.at()` and  handle exception or validate size.

Reply:
`getCargo` return vector of ptr. It could not have value at index. And you got an exception. Please handle exceptions, or validate if you actually can get cargo and this index.